pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1396,24 +1421,63 @@ func getDetachedDisks(oldDisks, newDisks []api.Disk) []api.Disk {
 func getAttachedDisks(oldDisks, newDisks []api.Disk) []api.Disk {
 	oldDiskMap := make(map[string]api.Disk)
 	for _, disk := range oldDisks {
-		file := getSourceFile(disk)
-		if file != "" {
-			oldDiskMap[file] = disk
-		}
+		oldDiskMap[disk.Target.Device] = disk
 	}
 	res := make([]api.Disk, 0)
 	for _, newDisk := range newDisks {
 		if !isHotplugDisk(newDisk) {
 			continue
 		}
-		if _, ok := oldDiskMap[getSourceFile(newDisk)]; !ok {
+		if _, ok := oldDiskMap[newDisk.Target.Device]; !ok {
 			// This disk got attached, add it to the list
 			res = append(res, newDisk)
 		}
 	}
 	return res
 }
 
+func isHotPlugDiskOrEmpty(disk api.Disk) bool {
+	return isHotplugDisk(disk) || getSourceFile(disk) == ""
+}
+
+func getUpdatedDisks(oldDisks, newDisks []api.Disk) []api.Disk {
```

Comment:
I would love to see unit tests only for this function, even if it is covered by the one who calls it. It is pretty complex

Reply:
done